### PR TITLE
Use custom node pools with --auto-mode

### PR DESCRIPTION
**Pull request by cartermckinnon**
*Description of changes:*

This PR adds a custom `NodePool` when `--auto-mode` is used, so that the existing `--instance-types` flag takes effect. This ensures that a consistent set of default instance types is used regardless of the compute type.

This also removes the SSH key logic controlled by `--generate-ssh-key`. This is not needed. I've left the flag in place so it can be removed from downstream jobs, but the flag is now a no-op.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: kubetest2/internal/deployers/eksapi/infra.go [hunk 1]
```diff
@@ -233,6 +246,53 @@ func (m *InfrastructureManager) deleteInfrastructureStack() error {
 	return nil
 }
 
+// deleteLeakedIntanceProfiles deletes any instance profiles to which the node role is attached,
+// because this will block node role deletion (and deletion of the infrastructure stack).
+// For example, when --auto-mode is used, an instance profile will be created for us and won't be deleted automatically with the cluster.
+func (m *InfrastructureManager) deleteLeakedInstanceProfiles(infra *Infrastructure) error {
```
Comment (ndbaker1) [inline kubetest2/internal/deployers/eksapi/infra.go:252]: nice catch

File: kubetest2/internal/deployers/eksapi/k8s.go [hunk 2]
```diff
@@ -9,35 +9,55 @@ import (
...
-	return kubernetes.NewForConfig(c)
+	return &k8sClient{
+		config:    config,
+		clientset: kubernetes.NewForConfigOrDie(config),
+		client:    util.Must(client.New(config, client.Options{})),
+	}, nil
 }
 
-func waitForReadyNodes(client *kubernetes.Clientset, nodeCount int, timeout time.Duration) error {
+func (k *k8sClient) waitForReadyNodes(nodeCount int, timeout time.Duration) error {
```
Comment (ndbaker1) [inline kubetest2/internal/deployers/eksapi/k8s.go:52]: nice ergonomic changes 😌

File: kubetest2/internal/deployers/eksapi/node.go [hunk 3]
```diff
@@ -177,17 +263,7 @@ func (m *nodeManager) createManagedNodegroup(infra *Infrastructure, cluster *Clu
...
-	if len(opts.InstanceTypes) > 0 {
-		input.InstanceTypes = opts.InstanceTypes
-	} else {
-		// managed nodegroups uses a t3.medium by default at the time of writing
-		// this only supports 17 pods, which can cause some flakes in the k8s e2e suite
-		defaultInstanceTypes, err := m.getValidDefaultInstanceTypesByEKSAMIType(input.AmiType)
-		if err != nil {
-			return fmt.Errorf("failed to get default instance types for AmiType: %s: %v", input.AmiType, err)
-		}
-		input.InstanceTypes = defaultInstanceTypes
```
Comment (ndbaker1) [inline kubetest2/internal/deployers/eksapi/node.go:189]: we still want `input.InstancesTypes` set right? i know the instance type resolving step was moved earlier in the process

File: kubetest2/internal/deployers/eksapi/k8s.go [hunk 4]
```diff
@@ -9,35 +9,55 @@ import (
...
+func newK8sClient(kubeconfigPath string) (*k8sClient, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
 	if err != nil {
 		return nil, err
 	}
-	return kubernetes.NewForConfig(c)
+	return &k8sClient{
+		config:    config,
+		clientset: kubernetes.NewForConfigOrDie(config),
+		client:    util.Must(client.New(config, client.Options{})),
```
Comment (ndbaker1) [inline kubetest2/internal/deployers/eksapi/k8s.go:48]: is this the only place the `Must` got used 🤣

**Review by ndbaker1 — COMMENTED** [review]
nice! one real comment

File: kubetest2/internal/deployers/eksapi/node.go (hunk 3 above)
Comment (cartermckinnon) [reply kubetest2/internal/deployers/eksapi/node.go:189]: Yep thx! fixed

File: kubetest2/internal/deployers/eksapi/k8s.go (hunk 4 above)
Comment (cartermckinnon) [reply kubetest2/internal/deployers/eksapi/k8s.go:48]: for now 😉

**Review by ndbaker1 — APPROVED** [review]

**Pull request merged**